src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -407,6 +408,21 @@ class RichRDDByteArray(val r: RDD[Array[Byte]]) extends AnyVal {
   }
 }
 
+class RichPairRDD[K, V](val r: RDD[(K, V)])
+                       (implicit kt: ClassTag[K], vt: ClassTag[V], ord: Ordering[K] = null) {
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)]): RDD[(K, (V, Option[W]))] = leftOuterJoinDistinct(other, defaultPartitioner(r, other))
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)], partitioner: Partitioner) = {
```

Comment:
I don't think we need leftOuterJoinDistinct in the FilterVariantsList code. I think you get the same result using leftOuterJoinDistinct as leftOuterJoin in this case. It's comparing Some(_) to None of the right RDD values and the left RDD values are always unique in our current schema (va, gs). 

I think the leftOuterJoinDistinct should only be used for outputting data to users as I am concerned that using it blindly could affect the results, especially when we have multiple values per variant key (this will happen with array data in importplink).